src/controls/UserEditInteraction.js

```diff
@@ -0,0 +1,218 @@
+import Interaction from 'ol/interaction/Interaction.js';
+import VectorLayer from 'ol/layer/Vector.js';
+import RenderFeature, { toFeature } from 'ol/render/Feature.js';
+import VectorSource from 'ol/source/Vector.js';
+import Draw from 'ol/interaction/Draw.js';
+import Modify from 'ol/interaction/Modify.js';
+import { defaultStyles } from '../constants.js';
+import {
+  getFeatureId,
+  getOlLayerForMbLayer,
+  isPointFeature,
+} from '../utils.js';
+
+/**
+ * @typedef {Object} Options
+ * @property {import('../YioMap.js').YioMap} yioMap
+ * @property {import('ol/Map.js').default} map ol map
+ */
+
+export default class UserSelectInteraction extends Interaction {
+  /** @type {import('../YioMap.js').YioMap} */
+  #yioMap = null;
+
+  /** @type {import('ol/Map.js').default} */
+  #map = null;
+
+  /**
+   * @type {VectorLayer} layer for temporary features of the draw / modify interactions
+   */
+  #editLayer = null;
+
+  getEditLayer() {
+    return this.#editLayer;
+  }
+
+  /**
+   * ol layer connected associated with the given mapbox `source-layer` (editLayer)
+   * @type {VectorLayer | import("ol/layer/VectorTile.js").default}
+   */
+  #olSourceLayer;
```

Comment:
Nitpicking I know, but I think `editSourceLayer` would be a better name for this one. I'd like to avoid the `ol` prefix.